mlrun/runtimes/base.py

```diff
@@ -315,102 +315,39 @@ def run(
             self.fill_credentials()
 
         if local:
-            if schedule is not None:
-                raise mlrun.errors.MLRunInvalidArgumentError(
-                    "local and schedule cannot be used together"
-                )
-            # allow local run simulation with a flip of a flag
-            command = self
-            if local_code_path:
-                project = project or self.metadata.project
-                name = name or self.metadata.name
-                command = local_code_path
-            return mlrun.run_local(
+            return self._run_local(
                 runspec,
-                command,
+                schedule,
+                local_code_path,
+                project,
                 name,
-                self.spec.args,
-                workdir=workdir,
-                project=project,
-                handler=handler,
-                params=params,
-                inputs=inputs,
-                artifact_path=artifact_path,
-                mode=self.spec.mode,
-                allow_empty_resources=self.spec.allow_empty_resources,
+                workdir,
+                handler,
+                params,
+                inputs,
+                artifact_path,
             )
 
-        if runspec:
-            runspec = deepcopy(runspec)
-            if isinstance(runspec, str):
-                runspec = literal_eval(runspec)
-            if not isinstance(runspec, (dict, RunTemplate, RunObject)):
-                raise ValueError(
-                    "task/runspec is not a valid task object," f" type={type(runspec)}"
-                )
-
-        if isinstance(runspec, RunTemplate):
-            runspec = RunObject.from_template(runspec)
-        if isinstance(runspec, dict) or runspec is None:
-            runspec = RunObject.from_dict(runspec)
+        runspec = self._resolve_runspec_runtime(runspec)
```

Comment:
Since it's the whole run object I guess let's go with
```suggestion
        run = self._create_run_object(runspec)
```